### ISSUE TITLE
Tree shaking is not working at all

### ISSUE THREAD

**Issue by xe4me** [AngularClass/angular-starter#1570]
**Note: for support questions, please use one of these channels:** [Chat: AngularClass.slack](http://angularclass.com/member-join/) or [Twitter: @AngularClass](https://twitter.com/AngularClass)

* **I'm submitting a  bug ...**
[*] bug report

* **What is the current behavior?**
Unused modules are not shaken out and they'll end up in main bundle both for AOT and JIT.

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal 

I've created a folder called `files` in the app directory as bellow : 

app.module.ts
app.component.ts
files : 
     -------index.ts
     -------SHOULD_BE_SHAKEN.TS
     -------SHOULD_BE_THERE.TS

I have two files , which I'm exporting them inside the `index.ts` of files folder as bellow : 

```
export { SHOULD_BE_THERE } from './SHOULD_BE_THERE'
export { SHOULD_BE_SHAKEN } from './SHOULD_BE_SHAKEN'

```
And inside my app.module.ts , I'm only importing and using SHOULD_BE_THERE:

```
import { SHOULD_BE_THERE } from './files';
@NgModule( {
    bootstrap    : [ AppComponent ],
    declarations : [
        AppComponent
    ],
    imports      : [ 
        BrowserModule,
        FormsModule,
        HttpModule,
    ],
    providers    : [
        SHOULD_BE_THERE
    ]
} )
export class AppModule {
    constructor ( service : SHOULD_BE_THERE ) {
    }
}
```

And here is inside both those services : 

`SHOULD_BE_SHAKEN.ts`

```
import {
    Injectable,
    EventEmitter
} from '@angular/core';
@Injectable()
export class SHOULD_BE_SHAKEN {
    constructor () {
        let m = "SHOULD_BE_SHAKEN";
        alert(m);
    }
}
```

`SHOULD_BE_THERE`
```
import {
    Injectable,
    EventEmitter
} from '@angular/core';
@Injectable()


@Injectable()
export class SHOULD_BE_THERE {
    constructor () {
        let m = "SHOULD_BE_THERE";
        alert(m);
    }
}


@Injectable()
export class SHOULD_BE_SHAKEN2 {
    constructor () {
        let m = "SHOULD_BE_SHAKEN2";
        alert(m);
    }
}
```

And then I'm running `npm run build:prod or build:aot:prod` expecting not to find the `SHOULD_BE_SHAKEN` and `SHOULD_BE_SHAKEN2` inside the `main bundle` , but as you can see both of them ended up there : 

Inside the main bundle (minified of course ) : 

```
{function t(){var t="SHOULD_BE_SHAKEN";alert(t)}return t})();r=_.b([n.i(i.a)(),_.c("design:paramtypes",[])],r)},357:function(t,e,n){"use strict";var _=n(229);n.d(e,
```

```
{var t="SHOULD_BE_SHAKEN2";alert(t)}return t})();o=_.b([n.i(i
```

* **What is the expected behavior?**

SHOULD_BE_SHAKEN and SHOULD_BE_SHAKEN2 not to be bundled.

* **Please tell us about your environment:**
Mac os, latest, Siera

And I've got the : 

`  "name": "angular2-webpack-starter",
    "version": "5.4.1",
`

This is really crucial for us as the bundle ends up being 3mg !

Thanks and Cheers

**Comment by admosity** [comment]
Maybe webpack/webpack#2867

**Comment by dereklin** [comment]
@xe4me Does this problem still exist?  Does it make any difference if you import like this?:
```
import { SHOULD_BE_THERE } from './files/SHOULD_BE_THERE';
```

**Comment by MarkPieszak** [comment]
Looks like it's a feature that will be fixed in 2.4 webpack as per @admosity issue link
